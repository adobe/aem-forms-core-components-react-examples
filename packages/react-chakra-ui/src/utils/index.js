import React from 'react';
import { FormContext } from '@aemforms/af-react-renderer';
import { createFormInstance } from '@aemforms/af-core';
import { ChakraProvider } from '@chakra-ui/react'
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';

export const DEFAULT_ERROR_MESSAGE = 'There is an error in the field';

export const createForm = (field) => {
  const formJson = {
    items: [field]
  };
  return createFormInstance(formJson);
};

export const Provider = (form, mappings, locale, dictionaries) => props => {
  const c = {
    form,
    mappings,
    modelId: '$form',
    refMap: {}
  };
  const { children } = props;
  return (
    <ChakraProvider>
      <IntlProvider locale={locale} messages={dictionaries}>
        <FormContext.Provider value={c}>{children}</FormContext.Provider>
      </IntlProvider>
    </ChakraProvider>
  );
};

export const renderComponent = function (Component) {
  const test = (field, operation) => {
    const form = createForm(field);
    if (operation) {
      operation(form, form.items[0]);
    }
    const e = form.items[0].getState();
    let component = <Component {...e} />;
    const wrapper = Provider(form);
    const renderResponse = render(component, { wrapper });
    return {
      renderResponse,
      form,
      element: form?.items[0],
      component
    };
  };
  return test;
};
