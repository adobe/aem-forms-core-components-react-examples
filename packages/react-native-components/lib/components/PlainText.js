"use strict";
/*************************************************************************
* ADOBE CONFIDENTIAL
* ___________________
*
* Copyright 2023 Adobe
* All Rights Reserved.
*
* NOTICE: All information contained herein is, and remains
* the property of Adobe and its suppliers, if any. The intellectual
* and technical concepts contained herein are proprietary to Adobe
* and its suppliers and are protected by all applicable intellectual
* property laws, including trade secret and copyright laws.
* Dissemination of this information or reproduction of this material
* is strictly forbidden unless prior written permission is obtained
* from Adobe.

* Adobe permits you to use and modify this file solely in accordance with
* the terms of the Adobe license agreement accompanying it.
*************************************************************************/
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const react_native_1 = require("react-native");
const native_base_1 = require("native-base");
const react_native_render_html_1 = __importDefault(require("react-native-render-html"));
const withRuleEngine_1 = __importDefault(require("../shared/withRuleEngine"));
const PlainTextComponent = function (props) {
    const { width } = (0, react_native_1.useWindowDimensions)();
    if (props.richText) {
        return (react_1.default.createElement(react_native_render_html_1.default, { contentWidth: width, source: { html: props === null || props === void 0 ? void 0 : props.value } }));
    }
    else {
        return (react_1.default.createElement(native_base_1.Text, null, props.value || ''));
    }
};
exports.default = (0, withRuleEngine_1.default)(PlainTextComponent);
