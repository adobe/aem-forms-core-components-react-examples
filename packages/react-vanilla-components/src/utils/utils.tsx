export const formatBytes = (bytes: number, decimals = 0) => {
    if (!+bytes) {
      return '0 Bytes';
    }
    const k = 1024;
    const dm = decimals < 0 ? 0 : decimals;
    const sizes = ['bytes', 'kb', 'mb', 'gb', 'tb', 'pb'];
    const i = Math.floor(Math.log(bytes) / Math.log(k));

    return `${parseFloat((bytes / Math.pow(k, i)).toFixed(dm))} ${sizes[i]}`;
  };

  export const syncAriaDescribedBy = (id: string, tooltip?: string, description?: string, error?: string) => {
    const descriptions = [];

    if (description) {
        descriptions.push(`${id}__longdescription`);
    }
    if (tooltip) {
        descriptions.push(`${id}__shortdescription`);
    }
    if (error) {
        descriptions.push(`${id}__errormessage`);
    }

    return descriptions.join(' ');
};