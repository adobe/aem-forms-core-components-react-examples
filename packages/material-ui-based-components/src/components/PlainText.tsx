import React from 'react';
import { withRuleEngine, richTextString } from '../shared/withRuleEngine';
import { PROPS } from '../utils/types';
import { Typography } from '@mui/material';


const PlainTextComponent = (props: PROPS) => {
    const { value, richText } = props;

    return (
        <Typography gutterBottom>{richText ? richTextString(value) : value}</Typography>
    );
};

export default withRuleEngine(PlainTextComponent);