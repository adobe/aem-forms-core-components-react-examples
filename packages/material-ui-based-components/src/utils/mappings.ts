import Form from '../components/Form';
import Panel from '../components/Panel';
import TextField from '../components/TextField';
import Date from '../components/Date';
import PlainText from '../components/PlainText';
import Title from '../components/Title';
import Image from '../components/Image';
import Rating from '../components/Rating';
import Slider from '../components/Slider';
import TabsVerticalComp from '../components/TabsVerticalComp';
import TabsHorizontalComp from '../components/TabsHorizontalComp';

export default {
  form: Form,
  panel: Panel,
  'text-input': TextField,
  'date-input': Date,
  'text-input': TextField,
  'plain-text': PlainText,
  'custom:title': Title,
  image: Image,
  'text-input': TextField,
  'custom:rating':Rating,
  'custom:slider':Slider
  'text-input': TextField,
  'custom:vertical-tab': TabsVerticalComp,
  'custom:horizontal-tab': TabsHorizontalComp
};