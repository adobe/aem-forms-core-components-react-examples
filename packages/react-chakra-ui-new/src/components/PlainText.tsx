/*************************************************************************
* ADOBE CONFIDENTIAL
* ___________________
*
* Copyright 2023 Adobe
* All Rights Reserved.
*
* NOTICE: All information contained herein is, and remains
* the property of Adobe and its suppliers, if any. The intellectual
* and technical concepts contained herein are proprietary to Adobe
* and its suppliers and are protected by all applicable intellectual
* property laws, including trade secret and copyright laws.
* Dissemination of this information or reproduction of this material
* is strictly forbidden unless prior written permission is obtained
* from Adobe.

* Adobe permits you to use and modify this file solely in accordance with
* the terms of the Adobe license agreement accompanying it.
*************************************************************************/

import React from 'react';
import { withRuleEngine, richTextString } from '../shared/withRuleEngine';
import { PROPS } from '../utils/type';
import { Box } from '@chakra-ui/react';


const PlainTextComponent = (props: PROPS) => {
    const { value, richText } = props;
    return (
        <Box>{richText ? richTextString(value) : value}</Box>
    );
};

export default withRuleEngine(PlainTextComponent);