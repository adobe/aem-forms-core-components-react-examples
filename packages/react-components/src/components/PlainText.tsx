// *******************************************************************************
//  * Copyright 2022 Adobe
//  *
//  * Licensed under the Apache License, Version 2.0 (the “License”);
//  * you may not use this file except in compliance with the License.
//  * You may obtain a copy of the License at
//  *
//  *     http://www.apache.org/licenses/LICENSE-2.0
//  *
//  * Unless required by applicable law or agreed to in writing, software
//  * distributed under the License is distributed on an “AS IS” BASIS,
//  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  * See the License for the specific language governing permissions and
//  * limitations under the License.

//  * The BEM markup is as per the AEM core form components guidelines.
//  * LINK- https://github.com/adobe/aem-core-forms-components/blob/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/text/v1/text/text.html
//  ******************************************************************************

import React from 'react';
import {withRuleEngine, richTextString} from '../utils/withRuleEngine';
import { PROPS } from '../utils/type';

const PlainText = (props: PROPS) => {
   const { value, id, visible } = props; 
   return (
      <div className='text base'>
        <div data-cmp-is='adaptiveFormText' id={id} className='cmp-adaptiveform-text' data-cmp-visible={visible}>
            {props?.richText ? richTextString(value) : value}
        </div>
      </div>
   );
};

export default withRuleEngine(PlainText);