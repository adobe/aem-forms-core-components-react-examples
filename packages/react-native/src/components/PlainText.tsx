import { State, FieldJson } from '@aemforms/af-core';
import { useRenderer } from '@aemforms/af-react-renderer';
import { Text } from 'native-base';
import { combineConvertors, baseConvertor, plainTextConvertor } from '../utils/mappers';

const PlainTextComponent = function (props: State<FieldJson>) {
  const renderedComponent = useRenderer(props, Text, combineConvertors(baseConvertor, plainTextConvertor));
  return renderedComponent;
};

export default PlainTextComponent;