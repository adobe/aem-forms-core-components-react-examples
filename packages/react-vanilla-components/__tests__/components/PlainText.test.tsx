/*
 * Copyright 2023 Adobe, Inc.
 *
 * Your access and use of this software is governed by the Adobe Customer Feedback Program Terms and Conditions or other Beta License Agreement signed by your employer and Adobe, Inc.. This software is NOT open source and may not be used without one of the foregoing licenses. Even with a foregoing license, your access and use of this file is limited to the earlier of (a) 180 days, (b) general availability of the product(s) which utilize this software (i.e. AEM Forms), (c) January 1, 2023, (d) Adobe providing notice to you that you may no longer use the software or that your beta trial has otherwise ended.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL ADOBE NOR ITS THIRD PARTY PROVIDERS AND PARTNERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import PlainText from '../../src/components/PlainText';
import { renderComponent } from '../utils';

const field = {
  name: "text",
  fieldType: "plain-text",
  value: "Plain Text",
  visible: true
};
const helper = renderComponent(PlainText);

describe('Plain Text', () => {

  test('should render with value', () => {
    const f = {
      ...field,
    };
    const { renderResponse } = helper(f);
    const form = renderResponse.getByText(field.value);
    expect(form).toBeTruthy();
  });

  test('should not render without value', () => {
    const f = {
      ...field,
      value: null
    };
    const { renderResponse } = helper(f);
    const form = renderResponse.queryByText(field.value);
    expect(form).toBeNull();
  });

})