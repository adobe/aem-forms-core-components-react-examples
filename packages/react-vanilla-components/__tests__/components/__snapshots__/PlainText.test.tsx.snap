// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plain Text Snapshot 1`] = `
<div>
  <div
    class="text base"
  >
    <div
      class="cmp-adaptiveform-text"
      data-cmp-is="adaptiveFormText"
      data-cmp-visible="true"
      id="F2c8pJsL5bM"
    >
      Plain Text
    </div>
  </div>
</div>
`;
